components/base-node.js

```diff
@@ -44,6 +46,18 @@ module.exports = {
    },
 
   /**
+   * Standard component error handling - send a message to the error port and throw Error
+   *
+   * @param message message to be sent to error port
+   * @param errorPort handle to the component's error port
+   */
+   handleError: function( message, errorPort ) {
```

Comment:
The point is to centralize error handling so it is easy to modify the algorithm used.  If you are seeing errors reported twice, do you have the configuration hooked to an output component?    David wants to get away from using the noflo component output and do logging without it to simplify the edges.